### DISCUSSION TITLE
server comp returns empty data array (postman call and useeffect calls return data ok)

### DISCUSSION THREAD

**Discussion by ThiagoUnder** [supabase/supabase#14970] [Questions]
Im following the new youtube playlist on auth but I am having problems receiving data on server comp, even without RLS for tests. 

```
export default async function Home() {
     const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL!, process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!);
  const response = await supabase.from("notes").select();
```
gives me back:
{ "error": null, "data": [], "count": null, "status": 200, "statusText": "" }

The same client config but using a useEffect for the call:
```
  const supabase = createClient(process.env.NEXT_PUBLIC_SUPABASE_URL!, process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!);

  useEffect(() => {
    async function fetchNotes() {
      const { data, error } = await supabase.from("notes").select();
      if (error) console.error(error);
      console.log("the notes are: ", data);
    }
    fetchNotes();
  }, [supabase]);
```
gives me back the correct data array.
A Postman GET request to https://phkuogfrunoqgzsdvybb.supabase.co/rest/v1/notes also gives me back the correct array.

What am I missing?

**Comment by GaryAustin1** [comment]
Try adding a filter to the call. Several users on Supabase Discord are having issues with next.js caching the fetches.   Adding a filter would see if it busts the cache and show that is likely your issue. I don’t use next.js so not really following this closely, but you might also search the Discord.

**Reply by ThiagoUnder** [reply]
Ive added this
```
const currentTime = Date.now() - 60000; // 1 minute in the past
  const response = await supabase.from("notes").select().gte("lastModified", currentTime);
```
to try and bust any caching issues but no luck

**Reply by GaryAustin1** [reply]
Usually the only other way this occurs, with no filter, is not meeting RLS.  I did not suggest that as you appear to have no user and just doing anon key in the examples.

**Reply by ThiagoUnder** [reply]
yes I disabled RLS to test. Also the client comp and postman return a data array just fine

**Reply by ThiagoUnder** [reply]
> Usually the only other way this occurs, with no filter, is not meeting RLS. I did not suggest that as you appear to have no user and just doing anon key in the examples.

sorry Im not sure if you meant "if the request doesnt meet RLS policies" or "RLS is actually enabled but the request doesnt meet its standards" in any case I can only test it disabled. I am actually trying to implement auth so right now I can only disable/enable RLS until I figure out exactly the sb auth. Maybe Ill use Express since with this transition to app route maybe not all details are ironed out yet
